Shared/Samples/Show realistic light and shadows/ShowRealisticLightAndShadowsView.swift

```diff
@@ -0,0 +1,144 @@
+    /// - Parameter value: The slider's value.
+    private func sliderValueChanged(toValue value: Float) {
+        // A DateComponents struct to encapsulate the minute value from the slider.
+        let dateComponents = DateComponents(second: Int(value))
+        let startOfToday = Date.now
+        let date = Calendar.current.date(byAdding: dateComponents, to: startOfToday)!
+        dateTimeText = DateFormatter.localizedString(from: date, dateStyle: .medium, timeStyle: .short)
+        sunDate = date
+    }
+}
+
+extension ShowRealisticLightAndShadowsView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// A scene with buildings.
+        let scene: ArcGIS.Scene = {
+            // Creates a scene layer from buildings REST service.
+            let buildingsURL = URL(string: "https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer")!
+            let buildingsLayer = ArcGISSceneLayer(url: buildingsURL)
+            // Creates an elevation source from Terrain3D REST service.
+            let elevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+            let elevationSource = ArcGISTiledElevationSource(url: elevationServiceURL)
+            let surface = Surface()
+            surface.addElevationSource(elevationSource)
+            let scene = Scene(basemapStyle: .arcGISTopographic)
+            scene.baseSurface = surface
+            scene.addOperationalLayer(buildingsLayer)
+            scene.initialViewpoint = Viewpoint(
+                latitude: .zero,
+                longitude: .zero,
+                scale: .nan,
+                camera: Camera(latitude: 45.54605, longitude: -122.69033, altitude: 500, heading: 162.58544, pitch: 72.0, roll: 0)
+            )
+            return scene
+        }()
+        
+        /// The range of possible date second values.
+        /// The range is 0 to 86,340 seconds ((60 seconds * 60 minutes * 24 hours)  - 60 seconds),
+        /// which means 12 am to 11:59 pm.
+        static var dateSecondValueRange: ClosedRange<Float> { 0...86340 }
+        
+        /// The number of seconds to represent 12 pm (60 seconds * 60 minutes * 12 hours).
+        static let dateSecondBeforeNoon: Float = 43170
+    }
+}
+
+private extension SceneView.SunLighting {
+    static var allCases: [Self] { [.lightAndShadows, .light, .off] }
+    
+    /// A human-readable label of the sun lighting mode.
+    var label: String {
+        switch self {
+        case .lightAndShadows:
+            return "Light And Shadows"
+        case .light:
+            return "Light only"
+        case .off:
+            return "No Light"
+        }
+    }
+}
+
+private extension Date {
+    static let now = Calendar.current.startOfDay(for: .now)
```

Comment:
```suggestion
    static let startOfDay = Calendar.current.startOfDay(for: .now)
```
Should it be `startOfDay`?